### PR TITLE
[Localization] MSBuild Rework

### PR DESCRIPTION
**Pull request by wachaudh**
Based of https://github.com/xamarin/xamarin-macios/pull/7834. Main difference is the localized strings were abstracted to a library and the MSBuild order was modified for `Xamarin.MacDev.Tasks.Core.csproj` to call the `UpdateXlf` target before a build.  The Makefile was also updated to copy the localization assemblies to the correct folders post-build.

### CODE REVIEW

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CoreMLCompilerTaskBase.cs [hunk 1]
```diff
@@ -224,10 +225,10 @@ public override bool Execute ()
 						if ((Compile (model, outputDir, log, partialPlist)) != 0)
 							return false;
 					} else {
-						Log.LogMessage (MessageImportance.Low, "Skipping `{0}' as the output file, `{1}', is newer.", model.ItemSpec, partialPlist);
+						Log.LogMessage (MessageImportance.Low, MSBStrings.M0119, model.ItemSpec, partialPlist);
 					}
 				}
-
+
```
Comment (rolfbjarne) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CoreMLCompilerTaskBase.cs]: Whitespace change.

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs [hunk 2]
```diff
@@ -9,6 +9,8 @@
 using Xamarin.MacDev.Tasks;
 using Xamarin.MacDev;
 using Xamarin.Utils;
+using Xamarin.Localization.MSBuild;
+
```
Comment (rolfbjarne) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]:
```suggestion
```

File: msbuild/Xamarin.Localization.MSBuild/xlf/MSBStrings.cs.xlf [hunk 3]
```diff
@@ -0,0 +1,1943 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (whitneyschmidt) [inline msbuild/Xamarin.Localization.MSBuild/xlf/MSBStrings.cs.xlf]: Curious - I assume this file is autogenerated?

Comment (wachaudh) [reply msbuild/Xamarin.Localization.MSBuild/xlf/MSBStrings.cs.xlf]: Yes, all `.xlf` are autogenerated.

**Pull request merged**